build.ps1

```diff
@@ -22,26 +19,151 @@ function ZipContent([string] $sourceDirectory, [string] $target)
   [IO.Compression.ZipFile]::CreateFromDirectory($sourceDirectory, $target)
 }
 
+function CrossGen([string] $runtime, [bool] $isSelfContained, [string] $publishTarget, [string] $privateSiteExtensionPath)
+{
+    Write-Host "publishTarget: " $publishTarget
+    Write-Host "privateSiteExtensionPath: " $privateSiteExtensionPath
+
+    $selfContained = Join-Path $publishTarget "self-contained"
+    $regular = Join-Path $publishTarget "regular"
+    $croosGen = "$publishTarget\download\crossgen\crossgen.exe"
+
+    #DownloadNupkg "https://dotnet.myget.org/F/dotnet-core/api/v2/package/runtime.$runtime.microsoft.netcore.jit/2.1.0-preview2-26316-09" @("runtimes\$runtime\native\clrjit.dll")  @("$publishTarget\download\clrjit")
+    #DownloadNupkg "https://dotnet.myget.org/F/dotnet-core/api/v2/package/runtime.$runtime.Microsoft.NETCore.Runtime.CoreCLR/2.1.0-preview2-26316-09"  @("tools\crossgen.exe")  @("$publishTarget\download\crossgen")
+
+    DownloadNupkg "https://www.nuget.org/api/v2/package/runtime.$runtime.Microsoft.NETCore.Jit/2.0.5" @("runtimes\$runtime\native\clrjit.dll")  @("$publishTarget\download\clrjit")
+    DownloadNupkg "https://www.nuget.org/api/v2/package/runtime.$runtime.Microsoft.NETCore.Runtime.CoreCLR/2.0.5"  @("tools\crossgen.exe")  @("$publishTarget\download\crossgen")
+    DownloadNupkg "https://www.nuget.org/api/v2/package/Microsoft.Build.Tasks.Core/15.1.1012" @("lib\netstandard1.3\Microsoft.Build.Tasks.Core.dll")  @("$selfContained")
+    DownloadNupkg "https://www.nuget.org/api/v2/package/Microsoft.Build.Utilities.Core/15.1.1012" @("lib\netstandard1.3\Microsoft.Build.Utilities.Core.dll")  @("$selfContained")
+
+    # Publish self-contained app with all required dlls for crossgen
+    dotnet publish .\src\WebJobs.Script.WebHost\WebJobs.Script.WebHost.csproj -r $runtime -o "$selfContained" -v q /p:BuildNumber=$buildNumber    
+
+    # For self contained we want to process only IL ddls
+    if ($isSelfContained) {
+        dotnet publish .\src\WebJobs.Script.WebHost\WebJobs.Script.WebHost.csproj -o $regular -v q /p:BuildNumber=$buildNumber
+    }
```

Comment:
This looks confusing -- you check "if selfcontained", but you're publishing to "regular", yet you've already published to "selfContained". Is a variable name wrong somewhere?

Reply:
we publish to "regular" as we want to crossgen only our custom dlls. 
So we have two folders: "Self-Contained" and "Regular".
We will be processing all dlls which are in "self-contained" folder and in "Regular" folder with output to "self-contained".